packages/core/src/InstanceSettings.ts

```diff
@@ -168,6 +175,93 @@ export class InstanceSettings {
 
 	private save(settings: Settings) {
 		this.settings = settings;
-		writeFileSync(this.settingsFile, JSON.stringify(settings, null, '\t'), 'utf-8');
+		writeFileSync(this.settingsFile, JSON.stringify(this.settings, null, '\t'), {
+			mode: this.enforceSettingsFilePermissions.enforce ? 0o600 : undefined,
+			encoding: 'utf-8',
+		});
+	}
+
+	private loadEnforeSettingsFilePermissionsFlag(): {
+		isSet: boolean;
+		enforce: boolean;
+	} {
+		const envVar = process.env.N8N_ENFORCE_SETTINGS_FILE_PERMISSIONS?.toLowerCase();
+		const isEnvVarSet = !!envVar;
```

Comment:
since `!!('true')` and `!!('false')` will both evaluate to `true`, we should change this to:

```suggestion
		const isEnvVarSet = envVar === 'true';
```

Even better option would be to create a `@Config` class instead, and benefit from the already existing (and tested) boolean config parsing code.

Reply:
> since `core` doesn't have a dependency to `@n8n/config`

One of the primary purposes of `@n8n/config` was to extract out all of backend config, so that all backend specific packages could use them. So, adding this dependency would be perfectly fine.

Reply:
👍 done